advanced networking/devspaces-vnet-template.json

```diff
@@ -0,0 +1,598 @@
+            ],
+            "identity": {
+              "type": "UserAssigned",
+              "userAssignedIdentities": {
+                "[parameters('identity')]": {}
+              }
+            },
+            "properties": {
+              "forceUpdateTag": "1",
+              "AzCliVersion": "2.0.80",
+              "timeout": "PT30M",
+              "arguments": "'[resourceGroup().name]'",
+              "scriptContent": "az aks get-credentials -n private-aks-cluster -g $1 -f /tmp/kube-config;base64Config=$(base64 /tmp/kube-config); echo {\"base64Config\": \"$base64Config\"} > $AZ_SCRIPTS_OUTPUT_PATH",
+              "cleanupPreference": "OnSuccess",
+              "retentionInterval": "P1D"
+            }
+        },
+        {
+            "type": "Microsoft.DevSpaces/controllers",
+            "apiVersion": "2019-04-01",
+            "name": "private_controller",
+            "location": "[resourceGroup().location]",
+            "dependsOn": [
+                "[resourceId('Microsoft.ContainerService/managedClusters', 'private-aks-cluster')]",
+                "[resourceId('Microsoft.Resources/deploymentScripts', 'runBashWithOutputs')]"
+            ],
+            "sku": {
+                "name": "S1",
+                "tier": "Standard"
+            },
+            "properties": {
+                "targetContainerHostResourceId": "[resourceId('Microsoft.ContainerService/managedClusters', 'private-aks-cluster')]",
+                "targetContainerHostCredentialsBase64": "[reference('runBashWithOutputs').outputs.base64Config]",
+                "endpointType": "[parameters('endpointType')]"
+            }
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks/subnets",
+            "apiVersion": "2019-11-01",
+            "name": "[concat('private-vnet', '/aks-subnet')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/virtualNetworks', 'private-vnet')]"
+            ],
+            "properties": {
+                "addressPrefix": "10.0.5.0/24",
+                "serviceEndpoints": [
+                    {
+                        "service": "Microsoft.Sql",
+                        "locations": [
+                            "[resourceGroup().location]"
+                        ]
+                    },
+                    {
+                        "service": "Microsoft.AzureCosmosDB",
+                        "locations": [
+                            "*"
+                        ]
+                    },
+                    {
+                        "service": "Microsoft.KeyVault",
+                        "locations": [
+                            "*"
+                        ]
+                    },
+                    {
+                        "service": "Microsoft.Storage",
+                        "locations": [
+                            "[resourceGroup().location]",
+                            "westus2"
+                        ]
+                    }
```

Comment:
What are these?  Are they needed?

Reply:
Aks needs access to these things to get things deployed. If access to these resources is not possible then the creation of aks cluster fails.